llvm/lib/CodeGen/PreISelIntrinsicLowering.cpp

```diff
@@ -456,7 +456,11 @@ bool PreISelIntrinsicLowering::lowerIntrinsics(Module &M) const {
       break;
     case Intrinsic::exp:
       Changed |= forEachCall(F, [&](CallInst *CI) {
-        return lowerUnaryMathIntrinsicWithScalableVecArgAsLoop(M, CI);
+        // TODO: Check legality
+        if (!CI->getArgOperand(0)->getType()->isVectorTy()) {
```

Comment:
Without a legality check, doing this for all vectors is more likely to regress something in the fixed vector case

Reply:
Not sure if you saw my question above, but how do you do a legality check?